projects/igniteui-angular/src/lib/date-picker/date-picker.component.ts

```diff
@@ -508,990 +324,646 @@ export class IgxDatePickerComponent implements IDatePicker, ControlValueAccessor
-     *
-     * @hidden @internal
-     */
-    public handleSelection(date: Date): void {
-        if (this.value) {
-            date.setHours(this.value.getHours());
-            date.setMinutes(this.value.getMinutes());
-            date.setSeconds(this.value.getSeconds());
-            date.setMilliseconds(this.value.getMilliseconds());
-        }
-        const oldValue = this.value;
-        this.value = date;
+    /** @hidden @internal */
+    public registerOnChange(fn: any) {
+        this.onChangeCallback = fn;
+    }
+
+    /** @hidden @internal */
+    public registerOnTouched(fn: any) {
+        this.onTouchedCallback = fn;
+    }
 
-        this.emitValueChangeEvent(oldValue, this.value);
-        this.calendar.viewDate = date;
-        this.closeCalendar();
-        this.onSelection.emit(date);
+    /** @hidden @internal */
+    public setDisabledState?(isDisabled: boolean): void {
+        this.disabled = isDisabled;
     }
+    //#endregion
 
+    //#region Validator
     /** @hidden @internal */
-    public onOpenClick(event: MouseEvent) {
-        event.stopPropagation();
-        this.openDialog();
+    public registerOnValidatorChange(fn: any) {
+        this.onValidatorChange = fn;
     }
 
     /** @hidden @internal */
-    public onBlur(event, calcDate = true): void {
-        this._isInEditMode = false;
-        if (this.mode === InteractionMode.DropDown && calcDate) {
-            this.calculateDate(event.target.value, event.type);
+    public validate(control: AbstractControl): ValidationErrors | null {
+        const value = control.value;
+        const errors = {};
+        if (!value) {
+            Object.assign(errors, { value: true });
         }
+        if (value && this.disabledDates && isDateInRanges(value, this.disabledDates)) {
+            Object.assign(errors, { dateIsDisabled: true });
+        }
+        Object.assign(errors, DateTimeUtil.validateMinMax(value, this.minValue, this.maxValue));
 
-        if (this.collapsed) {
-            this._updateValidityOnBlur();
+        if (this.inputGroup && !this.inputGroup.isFocused) {
+            this.inputDirective.valid = this._ngControl.valid
+                ? IgxInputState.INITIAL
+                : IgxInputState.INVALID;
         }
+
+        return Object.keys(errors).length > 0 ? errors : null;
     }
+    //#endregion
 
     /** @hidden @internal */
-    public onFocus(): void {
-        this._isInEditMode = true;
-        if (this.value && this.invalidDate === '') {
-            this._transformedDate = this._getEditorDate(this.value);
+    public ngOnChanges(changes: SimpleChanges): void {
+        if (changes['locale']) {
+            this.inputFormat = DateTimeUtil.getDefaultInputFormat(this.locale || 'en')
+                || DateTimeUtil.DEFAULT_INPUT_FORMAT;
+        }
+        if (changes['displayFormat'] && this.dateTimeEditor) {
+            this.dateTimeEditor.displayFormat = this.displayFormat;
```

Comment:
If this is bound (it seems to be) than I don't see the point of it in ngOnChanges. Same for input format